nntrainer/tensor/blas_neon.cpp

```diff
@@ -2042,5 +2042,20 @@ void elementwise_vector_addition_neon_fp16(const unsigned int N,
   }
 }
 
+void squared_root_transform_inplace_neon(const unsigned int N, __fp16 *X) {
+  unsigned int i = 0;
+  for (; N - i >= 8; i += 8) {
+    float16x8_t x0_7 = vld1q_f16(&X[i]);
+    float16x8_t x0_7_sqrt = vsqrtq_f16(x0_7);
+    float16x8_t ones = vmovq_n_f16(1);
+    float16x8_t x0_7_sqrt_div = vdivq_f16(ones, x0_7_sqrt);
+    vst1q_f16(&X[i], x0_7_sqrt_div);
+  }
+  while (i < N) {
+    X[i] = (1 / std::sqrt(static_cast<float>(X[i])));
```

Comment:
I just wonder if there is no accuracy loss when we use sqrt with fp16? is it ok because we use vsqrtq_f16?

Reply:
As far as I am concerned, we do not need to apply so-called 'enhanced-fp16' (using inter-fp32 values rather than full-fp16) for all fp16 functions. We have to optimize b/w time-accuracy only when it comes to "dimension shrinking" or some specific cases proven by a lot of papers ( BN, LN, sofrmax, ... )

I could observe the same model output with rms_norm layer with this function at glance, so I thought we could rely on it.
However indeed we should track on it becase it is a 'normalization' layer though.